File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
        Stack<Character> stack=new Stack();
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='*'){
                if(!stack.isEmpty()){
                stack.pop();
            }}
            else{
                stack.push(s.charAt(i));
            }
        }
        StringBuilder result = new StringBuilder();
        while (!stack.isEmpty()) {
            result.append(stack.pop());
        }
        return result.reverse().toString();
    }
}

File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.py
class Solution:
    def removeStars(self, s: str) -> str:
        stack=[]
        for i in s:
            if i.isalpha():
                stack.append(i)
            elif i=='*':
                stack.pop()
            else:
                return ''
        st=[str(x) for x in stack]
        return ''.join(st)